tests/fakedroid.sh

```diff
@@ -0,0 +1,114 @@
+#!/usr/bin/env bash
+
+# This is a script to run "on-device" tests in CI, without the device.
+# Takes the bootstrap aarch64 zipball, unpacks, proots into it.
+# This won't catch all bugs, of course, but that's something.
+
+# Set up envvars, prepare directories:
+
+set -uexo pipefail
+
+QEMU_URL=https://github.com/multiarch/qemu-user-static/releases/download/v6.1.0-8/qemu-aarch64-static
+QEMU=.fakedroid/inj/qemu-aarch64
+INSTALLATION_DIR=/data/data/com.termux.nix/files/usr
+TARGET_HOME=/data/data/com.termux.nix/files/home
+
+mkdir -p .fakedroid/inj
+mkdir -p .fakedroid/env/{$INSTALLATION_DIR,$TARGET_HOME,n-o-d}
+
+export PROOT_TMP_DIR=.fakedroid/env/$INSTALLATION_DIR/proot/tmp
+export PROOT_L2S_DIR=.fakedroid/env/$INSTALLATION_DIR/proot/l2s
+mkdir -p $PROOT_TMP_DIR
+mkdir -p $PROOT_L2S_DIR
+
+PASSTHROUGH_VARS=''
+PASSTHROUGH_VARS+=" PROOT_TMP_DIR=$PROOT_TMP_DIR"
+PASSTHROUGH_VARS+=" PROOT_L2S_DIR=$PROOT_L2S_DIR"
+PASSTHROUGH_VARS+=" TERM=$TERM"
+PASSTHROUGH_VARS+=" HOME=$TARGET_HOME"
+PASSTHROUGH_VARS+=" USER=$USER"
+set +u
+[[ -n "$CACHIX_SIGNING_KEY" ]] && \
+    PASSTHROUGH_VARS+=" CACHIX_SIGNING_KEY=$CACHIX_SIGNING_KEY"
+set -u
+
+PROOT_ARGS=''
+PROOT_ARGS+=' -r .fakedroid/env'
+PROOT_ARGS+=" -q $QEMU"
+PROOT_ARGS+=" -w $TARGET_HOME"
+PROOT_ARGS+=" -b .fakedroid/env/$INSTALLATION_DIR/nix:/nix"
+PROOT_ARGS+=" -b .fakedroid/env/$INSTALLATION_DIR/bin:/bin"
+PROOT_ARGS+=" -b .fakedroid/env/$INSTALLATION_DIR/etc:/etc"
+PROOT_ARGS+=" -b .fakedroid/env/$INSTALLATION_DIR/tmp:/tmp"
+PROOT_ARGS+=" -b .fakedroid/env/$INSTALLATION_DIR/usr:/usr"
+PROOT_ARGS+=' -b /dev'
+PROOT_ARGS+=' -b /proc'
+PROOT_ARGS+=' -b /sys'
+PROOT_ARGS+=' --link2symlink'
+
+
+# Procure a static QEMU for user emulation and a proot with our patches:
+
+[[ -e .fakedroid/inj/qemu-aarch64 ]] || wget $QEMU_URL -O $QEMU
+chmod +x $QEMU
+
+PROOT=$(nix-build tests/proot-test.nix)/bin/proot
+
+
+# Do the first install if not installed yet:
+
+if [[ ! -e .fakedroid/env/$INSTALLATION_DIR/etc ||
+        -e .fakedroid/env/$INSTALLATION_DIR/etc/UNINITIALIZED ]]; then
+    # Build a zipball:
+    nix build --show-trace -f pkgs \
+        --argstr arch aarch64 \
+        --argstr nixOnDroidChannelURL file:///n-o-d/archive.tar.gz \
+        bootstrapZip -o .fakedroid/inj/nix-on-droid-aarch64
+    ZIPBALL=$(realpath .fakedroid/inj/nix-on-droid-aarch64/bootstrap-aarch64.zip)
+    # Unpack the zipball the way the Android app does it:
+    pushd .fakedroid/env/$INSTALLATION_DIR
+        unzip "$ZIPBALL"
+        chmod -R 700 .
```

Comment:
Why should every file be executable?

Reply:
That's not right, I think I was aiming for u+rw here.

Reply:
Why would that be needed in the first place?

Reply:
Because unzip unpacks with -r-xr-xr-x: https://github.com/t184256/nix-on-droid/runs/4497615108?check_suite_focus=true#step:5:6223

Reply:
Ah right makes sense, it was build from stuff coming from /nix/store.

Reply:
Added some comment, removed second chmod.